tests/conftest.py

```diff
@@ -325,10 +327,17 @@ def _get_contract(
     *args,
     override_opt_level=None,
     input_bundle=None,
+    has_transient_storage=False,
     **kwargs,
 ):
     settings = Settings()
     settings.optimize = override_opt_level or optimize
+
+    if has_transient_storage and not version_check(begin="cancun"):
```

Comment:
instead of passing a parameter, let's either check the error message, or have a more specific exception type for transient storage

Reply:
how about this?
```
class UnsupportedTransientStorageException(EvmVersionException):
    """Transient storage is not supported for the active EVM ruleset."""
```

Reply:
sure, but let's do a bit shorter: `TransientStorageException(EvmVersionException)`